### PR TITLE
JAMES-2586 - JMAP Upload - cleanup, optimize, fix unstable test upload usage 

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

File: server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/upload/UploadServiceContract.scala [hunk 1]
```diff
@@ -108,6 +108,7 @@ trait UploadServiceContract {
         .block())
 
     // Exceed 100 bytes limit
+    Thread.sleep(500)
```
Comment (quantranhong1999) [inline server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/upload/UploadServiceContract.scala:111]: Q: any chance we can use await?

Comment (vttranlina) [reply server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/upload/UploadServiceContract.scala:111]: It is a difference.
Here I want the `cleanupUploadIfNeeded` completed (it was called 10 times) before triggering "exceed".

File: server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala [hunk 2]
```diff
@@ -206,4 +206,17 @@
...
 
+   @Test
+   def deleteShouldReturnTrueWhenRowExists(): Unit = {
+     val uploadId: UploadId = SMono.fromPublisher(testee.upload(data(), CONTENT_TYPE, USER)).block().uploadId
+
+     assertThat(SMono.fromPublisher(testee.delete(uploadId, USER)).block()).isTrue
+   }
+
+   @Test
+   def deleteShouldReturnFalseWhenRowDoesExists(): Unit = {
```
Comment (Arsnael) [inline server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala:217]:
```suggestion
   def deleteShouldReturnFalseWhenRowDoesNotExist(): Unit = {
```

**Review by Arsnael — COMMENTED** [review]
Can we contribute the non postgres enhancement on jmap upload to master branch as well?

**Comment by vttranlina** [comment]
The CI failed looks like not related to this pr
I tried to test it on local, the test passed
image

```
org.apache.james.DistributedPostgresJamesServerTest

02:47:05.340 [ERROR] t.postgres:16.1 - Could not start container
org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*database system is ready to accept connections.*\s'
	at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)
	at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)
	at org.testcontainers.containers.PostgreSQLContainer.waitUntilContainerStarted(PostgreSQLContainer.java:147)
```

**Comment by vttranlina** [comment]
> Can we contribute the non postgres enhancement on jmap upload to master branch as well?

I've split it into the master branch here: https://github.com/apache/james-project/pull/2069

As it relates to PostgreSQL changes, I'll rebase this pull request later, after the master branch has been merged.

**Review by hungphan227 — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

File: server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/upload/UploadServiceContract.scala [hunk 3]
```diff
@@ -126,6 +127,7 @@ trait UploadServiceContract {
...
```
Comment (chibenwa) [inline server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/upload/UploadServiceContract.scala:130]: Why?

Comment (vttranlina) [reply server/data/data-jmap/src/test/scala/org/apache/james/jmap/api/upload/UploadServiceContract.scala:130]: answered above: https://github.com/apache/james-project/pull/2068#discussion_r1506928950

- Before Thread sleep, we invoked `cleanupUploadIfNeeded` 10 times. I want to it completed before invoked the 11th.

File: server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/upload/UploadDAO.java [hunk 4]
```diff
@@ -171,6 +171,7 @@ public UploadDAO(CqlSession session, BlobId.Factory blobIdFactory) {
         this.delete = session.prepare(deleteFrom(TABLE_NAME)
             .whereColumn(USER).isEqualTo(bindMarker(USER))
             .whereColumn(ID).isEqualTo(bindMarker(ID))
+            .ifExists()
```
[the remark this answers is not in the document]
Comment (vttranlina) [reply server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/upload/UploadDAO.java:174]: I updated it on the master pr, https://github.com/apache/james-project/pull/2069
Will re-fixup this pr later

**Comment by vttranlina** [comment]
- [ ] todo rebase after:

- https://github.com/linagora/james-project/issues/5110

**Comment by Arsnael** [comment]
Why need to rebase after rebasing postgres branch with master btw?

**Comment by vttranlina** [comment]
> Can we merge this @vttranlina ?

This PR is using some code from master 
I'm waiting for the task https://github.com/linagora/james-project/issues/5110 done, then rebase this pr
Or I will cherry-pick a commit from the master

**Comment by vttranlina** [comment]
> Or I will cherry-pick a commit from the master

I did cherry-pick commits from master
https://github.com/apache/james-project/pull/2068/commits/869ee619231a93ca59445b6b5940a787c2a9f2cd
https://github.com/apache/james-project/pull/2068/commits/3e76f95bebed6f093e937a97cb6451df22944c26
(and resolved conflict)

**Pull request merged**